### PR TITLE
GraphQL Mesh

### LINKED ISSUE #239: Look into adding a GraphQL Mesh implementation
Look into creating a Federation test implementation that leverages [GraphQL Mesh](https://github.com/urigo/graphql-mesh).

### PR DESCRIPTION
**Pull request by enisdenjo**
Closes #239 

Add [GraphQL Mesh](https://www.the-guild.dev/graphql/mesh) implementation leveraging the [Apollo Federation Transform](https://www.the-guild.dev/graphql/mesh/docs/transforms/federation) for a fully compatible federation subgraph.

We start with a [completely regular GraphQL Product schema](https://github.com/enisdenjo/apollo-federation-subgraph-compatibility/blob/graphql-mesh-transform/implementations/graphql-mesh/products.graphql) and use [Mesh transformations](https://github.com/enisdenjo/apollo-federation-subgraph-compatibility/blob/graphql-mesh-transform/implementations/graphql-mesh/.meshrc.yml) to get the [required Apollo Federated schema](https://github.com/apollographql/apollo-federation-subgraph-compatibility#products-schema-to-be-implemented-by-library-maintainers).

### CODE REVIEW

**Review by dariuszkuc — CHANGES_REQUESTED** [review]
Hello 👋 
Thanks for the PR! Since Mesh is a gateway to expose multiple data sources as GraphQL - can we move it under `Other Solutions` section (just renamed `hosted subgraphs` section to `other solutions`)? We'll have to rethink how we will structure the results table as more "other" solutions are added but I think it is the best place to add it for now.

**Comment by enisdenjo** [comment]
@dariuszkuc done!

File: implementations/graphql-mesh/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fullName: GraphQL Mesh
+language: JavaScript / TypeScript
```
Comment (dariuszkuc) [inline implementations/graphql-mesh/metadata.yaml:2]:
```suggestion
language: Other Solutions
```

**Review by dariuszkuc — APPROVED** [review]

**Comment by dariuszkuc** [comment]
Thanks!

**Pull request merged**